### PR TITLE
[MLU] optimize matmul_grad_v2 dy (B,M,K)*(K,N) for better performance

### PR DESCRIPTION
**Pull request by cambriconhsq**
### PR types
Performance optimization 

### PR changes
OPs

### Describe
optimize matmul_grad_v2 case (B,M,K)*(K,N) for better performance

### CODE REVIEW

**Comment by Ivorfeng** [comment]
LGTM

**Comment by cambriconhsq** [comment]
image

File: paddle/fluid/operators/matmul_v2_op_mlu.cc [hunk 1]
```diff
@@ -333,22 +364,30 @@ class MatMulGradV2MLUKernel : public framework::OpKernel<T> {
...
-      if (y_dims != y_bcast_dims) {
-        dy_temp.Resize(phi::make_ddim(y_bcast_dims));
-      } else {
-        dY->mutable_data<T>(ctx.GetPlace());
-        dy_temp.ShareDataWith(*dY);
-      }
-
-      if (trans_y) {
-        MatMulND<T>(ctx, dout_temp, x_temp, &dy_temp, true, trans_x);
+      // Case 3: [B, M, K] x [K, N] =  [B, M, N]  better performance
```
Comment (ShawnNew) [inline paddle/fluid/operators/matmul_v2_op_mlu.cc:367]: 有机会可以在这个log上说明一下，输出tensor(下一步reduce的输入)会超过cnnl的大小限制

Comment (cambriconhsq) [reply paddle/fluid/operators/matmul_v2_op_mlu.cc:367]: 已更新说明

**Review by ShawnNew — APPROVED** [review]
👍

**Review by ronny1996 — APPROVED** [review]
LGTM

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**